src/eip7594/eip7594.c

```diff
@@ -691,21 +785,13 @@ C_KZG_RET verify_cell_kzg_proof_batch(
     // Array allocations
     ////////////////////////////////////////////////////////////////////////////////////////////////
 
-    ret = new_fr_array(&commitment_weights, num_commitments);
-    if (ret != C_KZG_OK) goto out;
     ret = new_fr_array(&r_powers, num_cells);
     if (ret != C_KZG_OK) goto out;
-    ret = new_fr_array(&weighted_powers_of_r, num_cells);
-    if (ret != C_KZG_OK) goto out;
-    ret = new_fr_array(&weights, num_cells);
-    if (ret != C_KZG_OK) goto out;
-    ret = new_g1_array(&commitments_g1, num_commitments);
-    if (ret != C_KZG_OK) goto out;
     ret = new_g1_array(&proofs_g1, num_cells);
     if (ret != C_KZG_OK) goto out;
 
     ////////////////////////////////////////////////////////////////////////////////////////////////
-    // Compute random linear combination of the proofs
+    // Compute powers of r, and extract KZG proofs out of input bytes
```

Comment:
I think we should move the proof conversions to the next section.

Reply:
Also, these section header comments aren't as valuable anymore. They were telling the reader what a particular group of function calls were doing. A single separate function does this better.

Reply:
> I think we should move the proof conversions to the next section.

I decided against this because `proofs_g1` is used in multiple sections below, so it's more of a global object like `r_powers`.

> Also, these section header comments aren't as valuable anymore. They were telling the reader what a particular group of function calls were doing. A single separate function does this better.

Agreed but refactoring the comments seems like an opinionated change that can happen in many different ways. Perhaps let's do it on another PR, since it seems largely orthogonal to this one?

Reply:
Ah I didn't notice that. And okay yeah, sounds good :+1: